protocol/news_feed_manager.go

```diff
@@ -0,0 +1,90 @@
+package protocol
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/mmcdole/gofeed"
+	gocommon "github.com/status-im/status-go/common"
+)
+
+// TODO replace with the real Status feed URL
+const STATUS_FEED_URL = "https://hnrss.org/frontpage"
+
+type FeedCallback func(item *gofeed.Item)
+
+type FeedParser interface {
+	ParseURL(url string) (*gofeed.Feed, error)
+}
+
+type NewsFeedManager struct {
+	url          string
+	parser       FeedParser
+	lastFetched  time.Time
+	interval     time.Duration
+	ctx          context.Context
+	cancel       context.CancelFunc
+	feedCallback FeedCallback
+}
+
+func NewNewsFeedManager(url string, parser FeedParser, pollingInterval time.Duration, fetchFrom time.Time, cb FeedCallback) *NewsFeedManager {
```

Comment:
5 arguments is almost too many, I think it would be nice to provide `Options` approach right away.
```go
opts := []newsfeed.Option{
    newsfeed.WithParser(parser),
    newsfeed.WithTiming(fetchFrom, pollInterval),
    // ...
}
```